src/platform/nrfconnect/MdnsImpl.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "lib/mdns/platform/Mdns.h"
+
+#include <platform/CHIPDeviceLayer.h>
+
+using namespace ::chip::DeviceLayer;
+
+namespace chip {
+namespace Mdns {
+
+CHIP_ERROR ChipMdnsInit(MdnsAsyncReturnCallback initCallback, MdnsAsyncReturnCallback errorCallback, void * context)
```

Comment:
I would actually add this file to all the Thread-based platform.. What do you think? It's super OpenThread specific and HW agnostic -> Maybe it suggests it should not be placed here.

Reply:
Maybe we should file an issue to turn `MdnsImpl` into a `MdnsManager` so that we could have a common `GenericMdnsManager_OpenThread` class from which other OT-based platforms could inherit? Having a single file for all OT platforms without the ability to override particular methods seems a bit too rigid. Thoughts?

Reply:
IMO both suggestions allow further development, but:

- Łukasz's idea gives us, as Damian said - rigid implementation, which can be improved in next steps, but at least we provide support for all Thread platforms.
- Damian's idea is about to not make more poorly designed code than we have to, as all idea of implementing SRP as backend for mDNS is temporary. 

Honestly I would follow with Łukasz idea, as in review nobody will accuse us about making nrf only implementation.

Reply:
Ok, go ahead :)

Reply:
Done